staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -456,6 +457,62 @@ func isDryRun(url *url.URL) bool {
 	return len(url.Query()["dryRun"]) != 0
 }
 
+type fieldValidationDirective int
+
+const (
+	ignoreFieldValidation fieldValidationDirective = iota
+	strictFieldValidation
+	warnFieldValidation
+)
+
+// fieldValidation checks if the fieldValidation query parameter is set on the request,
+// and if so ensures that the parameter is valid and that the request has a valid
+// media type, because the list of media types that support field validation are a subset of
+// all supported media types (only json and yaml supports field validation).
+func fieldValidation(req *http.Request) (fieldValidationDirective, error) {
+	supportedMediaTypes := []string{"application/json", "application/yaml"}
+	supported := false
+	contentType := req.Header.Get("ContentType")
+	// TODO: not sure if it is okay to assume empty content type is a valid one
+	if contentType != "" {
+		for _, v := range strings.Split(contentType, ",") {
+			t, _, err := mime.ParseMediaType(v)
+			if err != nil {
+				return ignoreFieldValidation, errors.NewBadRequest(fmt.Sprintf("could not parse media type: %v", v))
+			}
+			for _, mt := range supportedMediaTypes {
+				if t == mt {
+					supported = true
+					break
+				}
+			}
+		}
+		if !supported {
+			return ignoreFieldValidation, errors.NewBadRequest(fmt.Sprintf("fieldValidation parameter only supports media types types %v\n content type provided: %s", supportedMediaTypes, contentType))
+		}
+	}
+
+	validationParam := req.URL.Query()["fieldValidation"]
+	switch len(validationParam) {
+	case 0:
+		return ignoreFieldValidation, nil
+	case 1:
+		switch validationParam[0] {
```

Comment:
Does casing matter, if not, I'd do a strings.ToLower here

Reply:
Hmm, that’s a good question and one I’d pose to an API reviewer. I was using uppercase based on [this comment](https://github.com/kubernetes/enhancements/pull/2886#discussion_r702068352), but agree that it seems a little weird to require uppercase. I’ve added `strings.ToLower` for now. @lavalamp, do you know if query parameters are or should be case-sensitive?